src/test/java/org/apache/sling/feature/cpconverter/handlers/BundleEntryHandlerGAVTest.java

```diff
@@ -16,73 +16,168 @@
+        handler.setSlingInitialContentPolicy(SlingInitialContentPolicy.EXTRACT_AND_REMOVE);
+        handler.handle("/jcr_root/apps/gav/install/composum-nodes-config-2.5.3.jar", archive, entry, converter);
+        // verify no additional content package created (as it only contains the configuration which should end up in the feature model only)
+        Mockito.verify(converter, Mockito.never()).processContentPackageArchive(Mockito.any(), Mockito.isNull());
+        // modified bundle
+        Mockito.verify(featuresManager).addArtifact(null, ArtifactId.fromMvnId("com.composum.nodes:composum-nodes-config:2.5.3-cp2fm-converted"), null);
+        // need to use ArgumentCaptur to properly compare string arrays
+        Mockito.verify(featuresManager).addConfiguration(ArgumentMatchers.isNull(), ArgumentMatchers.eq("org.apache.sling.jcr.base.internal.LoginAdminWhitelist.fragment~composum_core_v2"), ArgumentMatchers.eq("/jcr_root/libs/composum/nodes/install/org.apache.sling.jcr.base.internal.LoginAdminWhitelist.fragment-composum_core_v2.config"), dictionaryCaptor.capture());
+        assertEquals("composum_core", dictionaryCaptor.getValue().get("whitelist.name"));
+        assertArrayEquals(new String[] {
+                "com.composum.nodes.commons",
+                "com.composum.nodes.pckgmgr",
+                "com.composum.nodes.pckginstall" }, (String[])dictionaryCaptor.getValue().get("whitelist.bundles"));
+    }
+    
+    @Test
+    public void testSlingInitialContentContainingConfigurationExtractAndKeep() throws Exception {
+        setUpArchive("/jcr_root/apps/gav/install/composum-nodes-config-2.5.3.jar", "composum-nodes-config-2.5.3.jar");
+        DefaultEntryHandlersManager handlersManager = new DefaultEntryHandlersManager();
+        converter.setEntryHandlersManager(handlersManager);
+        handler.setSlingInitialContentPolicy(SlingInitialContentPolicy.EXTRACT_AND_KEEP);
+        handler.handle("/jcr_root/apps/gav/install/composum-nodes-config-2.5.3.jar", archive, entry, converter);
+        // original bundle
+        Mockito.verify(featuresManager).addArtifact(null, ArtifactId.fromMvnId("com.composum.nodes:composum-nodes-config:2.5.3"), null);
+        // need to use ArgumentCaptur to properly compare string arrays
+        Mockito.verify(featuresManager).addConfiguration(ArgumentMatchers.isNull(), ArgumentMatchers.eq("org.apache.sling.jcr.base.internal.LoginAdminWhitelist.fragment~composum_core_v2"), ArgumentMatchers.eq("/jcr_root/libs/composum/nodes/install/org.apache.sling.jcr.base.internal.LoginAdminWhitelist.fragment-composum_core_v2.config"), dictionaryCaptor.capture());
+        assertEquals("composum_core", dictionaryCaptor.getValue().get("whitelist.name"));
+        assertArrayEquals(new String[] {
+                "com.composum.nodes.commons",
+                "com.composum.nodes.pckgmgr",
+                "com.composum.nodes.pckginstall" }, (String[])dictionaryCaptor.getValue().get("whitelist.bundles"));
+    }
```

Comment:
Should these tests be in the GAV test class? It seems they would make more sense in either a new dedicated test class or in the main BundleEntryHandlerTest class - I guess this is more nitpicking but it wouldn't be the first place I would look for them...

Reply:
BundleEntryHandlerTest is a parameterized test which is not suitable for this test. I can move them to a dedicated test class.

Reply:
I think that would be better (but again, I'm not saying you have to) - however, it would be good to at least add a bit more tests to make sure the bundles are stripped correctly etc.

Reply:
Done in https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/74/commits/e4acba48068e5582c610831b5427491fb7ddb103